internal/cmd/gentypes/main.go

```diff
@@ -173,6 +175,7 @@ import (
 				replace(btfID, "btf_id", "attach_btf_obj_id"),
 				replace(typeID, "attach_btf_id"),
 			},
+			[]string{"XlatedProgInsns", "MapIds", "LineInfo", "FuncInfo"},
```

Comment:
I worry that maintaining these by hand is going to be quite error prone. How about this:

1. Split `outputPatchedStruct` into `patchStruct` and `outputStruct` or similar.
2. Write a new function `findPointerTypes` which takes the patched result of `patchStruct` and iterates the members to find any where `Member.Type == pointer`. Return `Member.Name` as `[]string` as now.
3. Feed the patched struct and the `[]string` of the pointer types to `outputStruct`.

Reply:
Realized there's no distinct btf pointer type so I type asserted against `btf.Int` and checked if the `size == 8`

Reply:
True that there is no generally valid pointer, but for our purposes comparing against `pointer` should work fine.

Reply:
Interesting. Maybe I am missing something here but the `replace()` function takes in a pointer which is `pointer := &btf.Int{Size: 8}` and the `Type` field of the `Member` is an interface so type asserting against `btf.Pointer` doesn't match any member.

Reply:
I mean literally comparing the variables: `Member.Type == pointer`. This asserts that the variable Member.Type matches the `pointer` variable. Since both are an interface the comparison boils down to `types equal && pointers equal`.